### PR TITLE
Some table bug fixes

### PR DESCRIPTION
**Pull request by MattPutnam**
The scope of this has changed quite a bit...

While poking around with this issue, I found some bugs:
* The workflows list table was not refreshing with the filter buttons.  This turned out to be another faulty react optimization, so I gave it the current filter as a key.
* There was some bad logic with one of the submission details tables.  That has been fixed.

All in all, the component-will-receive-props logic in the table is no longer needed.

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs [hunk 1]
```diff
@@ -384,10 +384,6 @@
      (let [rows (react/call :get-filtered-data this)]
        (react/call :set-rows (@refs "body") (react/call :get-body-rows this rows))
        (react/call :set-num-rows-visible (@refs "paginator") (count rows))))
-   :component-will-receive-props
-   (fn [{:keys [props state refs]}]
-     (swap! state assoc :ordered-columns (create-ordered-columns (:columns props)))
-     (react/call :make-desynced (@refs "filterer")))
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs]: Maybe remove this method if it is no longer used.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs [hunk 2]
```diff
@@ -70,14 +70,8 @@
...
-       :columns [{:header "Data Entity" :starting-width 200 :sort-by :value
-                  :filter-by (fn [entity]
-                               (str (entity "entityType") " " (entity "entityName")))}
-                 {:header "Last Changed" :starting-width 280 :sort-by :value
-                  :content-renderer (fn [row-index date]
-                                      (let [m (js/moment date)]
-                                        (str (.format m "L [at] LTS") " ("
-                                          (.fromNow m) ")")))}
+       :columns [{:header "Data Entity" :starting-width 200 :sort-by :value}
+                 {:header "Last Changed" :starting-width 280 :sort-by :value}
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs]: Will this sort chronologically?

**Comment by dmohs** [comment]
:+1: with minor comments.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs (hunk 2 above)
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs]: Ack, you're right, it won't.  I'll revert.

**Pull request merged**